.github/workflows/increment-version-on-merge.yml

```diff
@@ -1,49 +1,28 @@
+# Workflow Code: AngryGoose_v3    DO NOT REMOVE
+
+# Note: This workflow should not be used with CI workflows that create their own releases, this workflow would conflict with that.
+
 name: Increment Version on Merge
-on: 
+on:
   pull_request:
     types: [closed]
 
 jobs:
   increment-version:
-    if: github.event.pull_request.merged == true && github.event.pull_request.base.ref == 'main'
-    
-    runs-on: [ubuntu-20.04]
+    if: github.event.pull_request.merged == true && github.event.pull_request.base.ref == 'main' # TODO:  verify default branch name
+
+    runs-on: ubuntu-latest # Force this to run on github-hosted runner by using a tag that does not exist on self-hosted runners
 
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
 
+      # See https://github.com/im-open/git-version for more details around how to increment major/minor/patch through commit messages
       - name: Increment the version
-        uses: actions/github-script@v4
-        with: 
-          github-token: ${{secrets.GITHUB_TOKEN}}
-          script: |
-            const util = require('util');
-            const exec = util.promisify(require('child_process').exec);
-
-            let newTag;
-            try {
-              ({ stdout: tag } = await exec(`git describe --tags --abbrev=0`));
-
-              tag = tag.replace('\n', '').trim();
-              core.info(`The latest tag is: ${tag}`);
-              const majorMinorPatch = tag.split('.');
-              const patch = parseInt(majorMinorPatch[2]) + 1;
-              newTag = `${majorMinorPatch[0]}.${majorMinorPatch[1]}.${patch}`
-              core.info(`The new tag is: ${newTag}`);
-            }
-            catch (error) {
-              newTag = 'v1.0.0';
-              core.info('An error occurred getting the tags for the repo.  It most likely does not have any tags to use. Defaulting to v1.0.0.');
-              core.info(error);
-            }
-
-            core.info(`Pushing tag '${newTag}' to the repository...`)
-            await github.git.createRef({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              ref: `refs/tags/${newTag}`,
-              sha: context.sha
-            });      
\ No newline at end of file
+        uses: im-open/git-version-lite@v2.0.0
+        with:
+          create-ref: true
+          github-token: ${{ secrets.GITHUB_TOKEN }} # Special per-job token generated by GH for interacting with the repo
```

Comment:
I don't want to add a comment for each of the comments (hehe) in this file because that's a lot of noise, but I'd think we could get rid of a few of them like this one.

Reply:
Do you think we need to remove all the comments?  They are what come from the template.

Reply:
I think the code comment and the `# See https://github.com/im-open/git-version ...` comment are probably good to keep. The others I can see arguments for being noisy. This particular one could have an argument made in favor of keeping, but I think anyone working in im-open's actions will have a good enough grasp of GitHub Actions to not need this explanation.